plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitManager.java

```diff
@@ -367,6 +371,18 @@ public CounterStat getHighMemorySplitSource()
                                 partitionBucketColumns,
                                 partitionBucketCount));
                     }
+                    if (isOptimizeSortedFileReads(session)) {
+                        List<SortingColumn> tableSortedColumns = bucketProperty.get().getSortedBy();
+                        List<SortingColumn> partitionSortedColumns = partitionBucketProperty.get().getSortedBy();
+                        if (!isSortingCompatible(tableSortedColumns, partitionSortedColumns)) {
```

Comment:
Should this be aware of name v/s name based table-partition mapping? I think it might conflict with schema evolution feature:

```
hive> CREATE TABLE xyz (a bigint, b bigint, c bigint) PARTITIONED BY (d bigint) 
		CLUSTERED BY (b) SORTED BY (c) INTO 32 BUCKETS STORED AS PARQUET;
presto> INSERT INTO xyz VALUES (5, 6, 7, 1), (8, 9, 10, 1);

hive> ALTER TABLE xyz REPLACE COLUMNS (a bigint, b bigint, cx bigint);
hive> ALTER TABLE xyz CLUSTERED BY (b) SORTED BY (cx) INTO 32 BUCKETS;

presto> SET SESSION hive.use_sorting_properties=true;
presto> SET SESSION hive.parquet_use_column_names=false;
presto> SET SESSION hive.partition_use_column_names=false;

presto> SELECT * FROM xyz;
// throws exception
```
Although this doesn't feel to be a common case,  we may want to handle it here. thoughts?

Reply:
Good catch, seems to be a genuine though uncommon scenario.
We can reproduce the same problem with just bucketing without sorting too
```
hive> CREATE TABLE xyz (a bigint, b bigint) PARTITIONED BY (d bigint) 
		CLUSTERED BY (b) INTO 32 BUCKETS STORED AS PARQUET;
presto> INSERT INTO xyz VALUES (5, 6, 1), (8, 9, 1);

hive> ALTER TABLE xyz REPLACE COLUMNS (a bigint, bx bigint);
hive> ALTER TABLE xyz CLUSTERED BY (bx) INTO 32 BUCKETS;

presto> SET SESSION hive.parquet_use_column_names=false;
presto> SET SESSION hive.partition_use_column_names=false;

presto> SELECT * FROM xyz;
// Hive table (default.xyz) bucketing (columns=[bx], buckets=32) is not compatible with partition (d=1) bucketing (columns=[b], buckets=32)
```
If acceptable, I would prefer to limit the scope of current PR and look into this as a separate issue as the problem is not unique to this functionality.
@sopel39 @findepi thoughts ?

Reply:
> If acceptable, I would prefer to limit the scope of current PR and look into this as a separate issue as the problem is not unique to this functionality.

I think it makes sense. I think we need explicit `io.trino.plugin.hive.HiveSessionProperties#isPartitionUseColumnNames` check here (+ test)

Reply:
> Should this be aware of name v/s index based table-partition mapping? I think it might conflict with schema evolution feature:

@phd3 @raunaqmorarka see @djsstarburst 's comment here https://github.com/trinodb/trino/pull/6479#issuecomment-768368188

Reply:
@raunaqmorarka fair enough. feel free to resolve this.